anymail/backends/amazon_ses.py

```diff
@@ -0,0 +1,239 @@
+from ..exceptions import AnymailAPIError, AnymailImproperlyInstalled
+from ..message import AnymailRecipientStatus
+from ..utils import get_anymail_setting
+
+try:
+    import boto3
+    from botocore.exceptions import BotoCoreError, ClientError, ConnectionError
+except ImportError:
+    raise AnymailImproperlyInstalled(missing_package='boto3', backend='amazon_ses')
+
+
+# boto3 has several root exception classes; this is meant to cover all of them
+BOTO_BASE_ERRORS = (BotoCoreError, ClientError, ConnectionError)
+
+
+# Work around Python 2 bug in email.message.Message.to_string, where long headers
+# containing commas or semicolons get an extra space inserted after every ',' or ';'
+# not already followed by a space. https://bugs.python.org/issue25257
+if Header("test,Python2,header,comma,bug", maxlinelen=20).encode() == "test,Python2,header,comma,bug":
+    # no workaround needed
+    def add_header(message, name, val):
+        message[name] = val
+
+else:
+    # workaround: custom Header subclass that won't consider ',' and ';' as folding candidates
+
+    class HeaderBugWorkaround(Header):
+        def encode(self, splitchars=' ', **kwargs):  # only split on spaces, rather than splitchars=';, '
+            return Header.encode(self, splitchars, **kwargs)
+
+    def add_header(message, name, val):
+        # Must bypass Django's SafeMIMEMessage.__set_item__, because its call to
+        # forbid_multi_line_headers converts the val back to a str, undoing this
+        # workaround. That makes this code responsible for sanitizing val:
+        if '\n' in val or '\r' in val:
+            raise BadHeaderError("Header values can't contain newlines (got %r for header %r)" % (val, name))
+        val = HeaderBugWorkaround(val, header_name=name)
+        assert isinstance(message, MIMEBase)
+        MIMEBase.__setitem__(message, name, val)
+
+
+class EmailBackend(AnymailBaseBackend):
+    """
+    Amazon SES Email Backend (using boto3)
+    """
+
+    esp_name = "Amazon SES"
+
+    def __init__(self, **kwargs):
+        """Init options from Django settings"""
+        super(EmailBackend, self).__init__(**kwargs)
+        # AMAZON_SES_CLIENT_PARAMS is optional - boto3 can find credentials several other ways
+        self.client_params = get_anymail_setting("client_params", esp_name=self.esp_name,
+                                                 kwargs=kwargs, allow_bare=False, default={})
+        # TODO: maybe add a setting for default configuration set?
+        #       (otherwise must use "AMAZON_SES_ESP_EXTRA": {"ConfigurationSetName": "my-default-set"})
+        self.client = None
+
+    def open(self):
+        if self.client:
+            return False  # already exists
+        try:
+            self.client = boto3.client("ses", **self.client_params)
+        except BOTO_BASE_ERRORS:
+            if not self.fail_silently:
+                raise
+
+    def close(self):
+        if self.client is None:
```

Comment:
looks like these couple of lines are unnecessary

Reply:
True, but I don't think they hurt anything, and they establish the pattern for closing the boto session if that ever makes sense... If (or when) there's a way to close a boto session, the comment immediately below would be replaced with something like `self.client.close()`. (Here's how that looks [in Requests-based backends](https://github.com/anymail/django-anymail/blob/feature_amazon_ses/anymail/backends/base_requests.py#L37-L46).)